Makefile

```diff
@@ -53,6 +51,16 @@ include make/testing_flags.mk
 
 DEV_TAGS := $(if ${tags},$(DEV_TAGS) ${tags},$(DEV_TAGS))
 
+LDFLAGS := -ldflags "-X $(PKG)/build.Commit=$(COMMIT) \
+	-X $(PKG)/build.CommitHash=$(COMMIT_HASH) \
+	-X $(PKG)/build.RawTags=$(shell echo ${tags} | sed -e 's/ /,/g')"
+DEV_LDFLAGS := -ldflags "-X $(PKG)/build.Commit=$(COMMIT) \
+	-X $(PKG)/build.CommitHash=$(COMMIT_HASH) \
+	-X $(PKG)/build.RawTags=$(shell echo $(DEV_TAGS) | sed 's/ /,/g')"
+ITEST_LDFLAGS := -ldflags "-X $(PKG)/build.Commit=$(COMMIT) \
+	-X $(PKG)/build.CommitHash=$(COMMIT_HASH) \
+	-X $(PKG)/build.RawTags=$(shell echo $(ITEST_TAGS) | sed 's/ /,/g')"
```

Comment:
Is this the way to do it, rather than using the build tags to conditionally compile variables in the code?

Reply:
i thought about it but i think it ends up being less brittle. we'd have to add a file in `build` for all tags ever passed to lnd. if they change or we add new ones its possible for `Tags()` to return the wrong value. i think it's easier just to set the variable with everything we are passing to the compiler.

Reply:
I was thinking of sliding it in with the conditionally compiled files that we already have.

Reply:
i don't think the engineering effort is worth it here. some build tags exist only outside of the build package, so we'd need to expose some sort of registration hook on init. then there's the issue of deduplicating and sorting the output so it's deterministic, all to end up with a solution that is still brittle as mentioned above.

Reply:
fwiw this also captures user-supplied tags, that we wouldn't be able to inspect if we do it via the file-based approach.

Reply:
I prefer the `Makefile` way too as it's all in a central place. And including user-specified tags could even be an useful feature if you want to tag your build for some reason. I'm voting for leaving it the way it is currently implemented in this PR.

Reply:
okay, leaving as is. i also cleaned this up a little to reduce code duplication